docs/src/architecture/domain-model.md

```diff
@@ -0,0 +1,184 @@
+* **Log Management**:
+    * allows viewing logs from the device
+      * lists available logs
+      * transfer individual logs from device to cloud
+    * a log is a text file, as e.g.
+      * log file(s) of the **domain application**[^1]
+      * log file(s) of **OS / Libs / Runtime**[^1]
+      * log file(s) of **thin-edge** it-self
+* **Device Monitoring**:
+    * collects metrics (e.g. cpu, memory and disk metrics) from the device and forward these to the IoT cloud
+    * allows monitoring the health of devices
+    * helps troubleshooting when problems with the device are reported
+
+[^1]: more details see appendix [Device Domain](#device-domain)
+
+## Child-Devices
+
+**thin-edge** facilitates IoT functionality to the device it is running on, as well as to devices that are connected to that device.
+  * the device **thin-edge** is running on is referred as the **main-device**
+    * **thin-edge** on the **main-device** establishes and manages all communication to the cloud
+  * all devices connected to the **main-device** are referred as **external child-devices** 
+  * each **external child-device** can be represented in the cloud with its individual **device twin**
+    * a unique **child-id** makes the association between each **external child-device** and it's **device twin**
+  * all **telemetry data** and **device management** functionality can appear in the context of the 
+    **external child-device's** individual **device twin**, or the **main-device's** **device twin**
+  * also _containers_ or _processes_ running on the **main-devices** can be handled like **external child-device**;
+    those are referered then as **logical child-devices**
+
+The figure below illustrates the child-device concept.
+
+![Device Concept](images/device-concept.svg)
+
+
+## Plugins
+
+**thin-edge** realizes **Device Management** based on **plugins**.
+  * a **plugin** can be an (external) executable (e.g. as the `c8y_configuration_plugin` for _configuration management_)
+    or a thin-edge built-in software component (e.g. as for _software management_)
+  * to realize **Device Management** functionality, a **plugin** accesses _resources_ and _services_ of the device, as e.g.
+      * _software management_ uses the device's _package manager_
+      * _configuration management_ reads/writes device's _configuration files_
+  * a **plugin** can facilitate **Device Management** functionality for the **main-device** and **external child-device**, but runs usually only on the **main-device**
+    * _resources_ and _services_ (e.g. file system, package manager, ...) of the **main-device** can be directly accessed by the **plugin**, as it runs on the **main-device**
+    * all _resources_ and _services_ of an **external child-device** are only remotely accessible for the **main-device**
+      * the **plugin** running on the **main-device** needs another software component that establish the remote access for those _resources_ and _services_
```

Comment:
The "other software component" (i.E. the child-device agent) must not necesarilly run on the child-device.

As you can see here (https://github.com/thin-edge/thin-edge.io/pull/1554/files/7ef7c3efe7ed174220c16c1415786eb3052ac7cb#diff-8fb5e0172ee785582c478280a6ef2f51c91d90961ee5f799ba1f5b212ae57c99R130) it can installed and executed on the **external child-device** or the **main-device**.